### PR TITLE
[FE] New Backend/Authentication (detect logout and already logged in)

### PR DESCRIPTION
**Pull request by Drakulix**
## What?

- Rebuild `Backend` as class to include current profile & login state
- Rebuild that state via Cookie and Request upon page load (reloading detects if still logged in or not)
- Handle logout by redirecting to SignIn page
- Model all Backend requests as functions of the Backend class
- Consistent returns as a Promise with value depending on the type of the Backend call (get, put, post, delete)
- Rewrite all components to use the new function, except DataTable, which still uses a manual get for the fetchURL property. (Should probably be replaced by a fetchFunction)

## Please test!

### CODE REVIEW

File: backend/__init__.py [hunk 1]
```diff
@@ -158,11 +160,14 @@ class EmailConverter(BaseConverter):
...
 SECURITY = Security(app, USER_DATASTORE)
 
+
 @SECURITY.login_manager.unauthorized_handler
 def handle_unauthorized_access():
     return make_response("Forbidden", 403)
 
 # internal function to append notifications to the given userlist
+
+
```
Comment (henderkes) [inline backend/__init__.py]: why? one new line before the comment and none after, please

**Review by henderkes — DISMISSED** [review]

File: backend/__init__.py (hunk 1 above)
Comment (Drakulix) [reply backend/__init__.py]: grr, that was autopep8, gimme a sec to fix this

**Comment by mharasic** [comment]
comment section is broken and not working
redirection to sign in is not working if a wrong uri is inserted

profile view delivers all projects
isAdmin is somehow broken (siderbar is evaluated correctly but all other places not...)

**Comment by Drakulix** [comment]
Should all be resolved now

**Review by mharasic — DISMISSED** [review]
Looks working

**Review by Rintel — APPROVED** [review]

**Pull request merged**